bundles/org.openhab.ui/web/src/pages/home/homecards-mixin.js

```diff
@@ -64,74 +64,102 @@ export default {
       } else if (item.metadata.semantics.config && item.metadata.semantics.config.hasLocation) {
         parent = (this.items.find((i) => i.name === item.metadata.semantics.config.hasLocation))
       }
-      item.modelPath = parent ? [...this.buildPathInModel(parent), parent] : []
+      if (parent && parent.semanticLoopDetector) {
+        this.loopError = `A a loop has been detected in the semantic model: ${parent.name} is both descendant and parent of ${item.name}`
+        throw this.loopError
+      }
+      item.semanticLoopDetector = true
+      item.modelPath = parent ? [...(this.buildPathInModel(parent, filteredItems)), parent] : []
+      delete item.semanticLoopDetector
+      item.parent = parent
+      item.children = []
+      item.locations = []
+      item.points = []
+      item.properties = []
+      item.equipment = []
+      item.equipmentOrPoints = []
+
+      if (parent) {
```

Comment:
here you add the item to the `filteredItems` only if it has a parent, which is surely wrong - there can be items at the root level that are part of the card model.

Reply:
Thanks for noticing this. I am actively testing this PR (by actually using it) on two installations with hundreds of items and so far I did not have any issue. But in my case there are no top level equipment, points or properties (everything is below the top-level location) so I did let this issue slip through.

I added some test items/groups and it appears to work fine now.